google/generativeai/types/model_types.py

```diff
@@ -183,16 +186,71 @@ class TuningExampleDict(TypedDict):
     output: str
 
 
-TuningExampleOptions = Union[TuningExampleDict, glm.TuningExample, tuple[str, str]]
+TuningExampleOptions = Union[TuningExampleDict, glm.TuningExample, tuple[str, str], list[str]]
 TuningDataOptions = Union[
-    glm.Dataset, Iterable[TuningExampleOptions]
-]  # TODO(markdaoust): csv, json, pandas, np
+    pathlib.Path, str, glm.Dataset, Mapping[str, Iterable[str]], Iterable[TuningExampleOptions]
+]  # TODO(markdaoust): json-URLS, gs:// URLS .. use etils.epath?
```

Comment:
If I want to support csv and json.. how do we handle these? Try to parse it as json, and then fallback to csv?

`[1,2,3]` could parse as either.

Expect try one format (csv?), and allow a `format="json"` argument?

Reply:
Good point.

An argument sounds good - otherwise we're still just making guesses or assumptions (e.g. the file handle is seekable, file fits in memory).

Just an idea anyway - if it's too ambiguous then let's not make it confusing.